{
  "active_lock_reason": null,
  "assignee": null,
  "assignees": [],
  "author_association": "MEMBER",
  "body": " /kind bug\r\n\r\n**What happened**:\r\nhttps://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/58320/pull-kubernetes-bazel-test/25992/\r\n```\r\nbazel test //pkg/cloudprovider/providers/azure:go_default_test\r\nE0116 08:02:57.246748      14 azure_loadbalancer.go:288] selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(true) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])\r\nE0116 08:02:57.345730      14 azure_loadbalancer.go:288] selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(false) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])\r\nE0116 08:02:58.790605      14 azure_blobDiskController.go:76] azureDisk - getAllStorageAccounts error: azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/--subscription-id--/resourceGroups/--resource-group--/providers/Microsoft.Storage/storageAccounts?api-version=2017-06-01: StatusCode=0 -- Original Error: adal: Refresh request failed. Status Code = '400'. Response body: {\"error\":\"invalid_request\",\"error_description\":\"AADSTS90002: Requested tenant identifier '--tenant-id--' is not valid.\\r\\nTrace ID: a6d564a2-94fc-432c-a8a0-23f026682500\\r\\nCorrelation ID: 925ff6f8-26c0-479c-bab9-2eff776147e4\\r\\nTimestamp: 2018-01-16 08:02:59Z\",\"error_codes\":[90002],\"timestamp\":\"2018-01-16 08:02:59Z\",\"trace_id\":\"a6d564a2-94fc-432c-a8a0-23f026682500\",\"correlation_id\":\"925ff6f8-26c0-479c-bab9-2eff776147e4\"}\r\nE0116 08:02:58.793439      14 azure_blobDiskController.go:76] azureDisk - getAllStorageAccounts error: storage.AccountsClient#ListByResourceGroup: Invalid input: autorest/validation: validation failed: parameter=resourceGroupName constraint=MinLength value=\"\" details: value length must be greater than or equal to 1\r\nE0116 08:02:58.795075      14 azure_blobDiskController.go:76] azureDisk - getAllStorageAccounts error: storage.AccountsClient#ListByResourceGroup: Invalid input: autorest/validation: validation failed: parameter=resourceGroupName constraint=MinLength value=\"\" details: value length must be greater than or equal to 1\r\nE0116 08:02:58.866717      14 azure_blobDiskController.go:76] azureDisk - getAllStorageAccounts error: azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/--subscription-id--/resourceGroups/--resource-group--/providers/Microsoft.Storage/storageAccounts?api-version=2017-06-01: StatusCode=0 -- Original Error: adal: Refresh request failed. Status Code = '400'. Response body: {\"error\":\"invalid_request\",\"error_description\":\"AADSTS90002: Requested tenant identifier '--tenant-id--' is not valid.\\r\\nTrace ID: b601ebed-6bfd-496c-a9da-ad8286413200\\r\\nCorrelation ID: 508095ad-5b26-4faf-8a0f-fef89cd8b0f0\\r\\nTimestamp: 2018-01-16 08:02:59Z\",\"error_codes\":[90002],\"timestamp\":\"2018-01-16 08:02:59Z\",\"trace_id\":\"b601ebed-6bfd-496c-a9da-ad8286413200\",\"correlation_id\":\"508095ad-5b26-4faf-8a0f-fef89cd8b0f0\"}\r\n==================\r\nWARNING: DATA RACE\r\nWrite at 0x00c42003b4b8 by goroutine 43:\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.setTestVirtualMachineScaleSets()\r\n      pkg/cloudprovider/providers/azure/azure_vmss_test.go:71 +0x10d5\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.TestGetInstanceIDByNodeName()\r\n      pkg/cloudprovider/providers/azure/azure_vmss_test.go:139 +0x366\r\n  testing.tRunner()\r\n      GOROOT/src/testing/testing.go:746 +0x16c\r\n\r\nPrevious read at 0x00c42003b4b8 by goroutine 40:\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.(*scaleSet).listScaleSetsWithRetry.func1()\r\n      pkg/cloudprovider/providers/azure/azure_vmss.go:406 +0x94\r\n  k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff()\r\n      vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:203 +0xb0\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.(*scaleSet).listScaleSetsWithRetry()\r\n      pkg/cloudprovider/providers/azure/azure_vmss.go:405 +0x2e9\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.(*scaleSet).updateCache()\r\n      pkg/cloudprovider/providers/azure/azure_vmss.go:106 +0x53\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.newScaleSet.func1()\r\n      pkg/cloudprovider/providers/azure/azure_vmss.go:96 +0x8d\r\n  k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1()\r\n      vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x6f\r\n  k8s.io/apimachinery/pkg/util/wait.JitterUntil()\r\n      vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xcd\r\n  k8s.io/apimachinery/pkg/util/wait.Until()\r\n      vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x5a\r\n\r\nGoroutine 43 (running) created at:\r\n  testing.(*T).Run()\r\n      GOROOT/src/testing/testing.go:789 +0x568\r\n  testing.runTests.func1()\r\n      GOROOT/src/testing/testing.go:1004 +0xa7\r\n  testing.tRunner()\r\n      GOROOT/src/testing/testing.go:746 +0x16c\r\n  testing.runTests()\r\n      GOROOT/src/testing/testing.go:1002 +0x521\r\n  testing.(*M).Run()\r\n      GOROOT/src/testing/testing.go:921 +0x206\r\n  main.main()\r\n      bazel-out/k8-fastbuild/bin/pkg/cloudprovider/providers/azure/go_default_test~/testmain.go:194 +0x367\r\n\r\nGoroutine 40 (running) created at:\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.newScaleSet()\r\n      pkg/cloudprovider/providers/azure/azure_vmss.go:92 +0x2a6\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.newTestScaleSet()\r\n      pkg/cloudprovider/providers/azure/azure_vmss_test.go:28 +0x34\r\n  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.TestGetInstanceIDByNodeName()\r\n      pkg/cloudprovider/providers/azure/azure_vmss_test.go:105 +0x65\r\n  testing.tRunner()\r\n      GOROOT/src/testing/testing.go:746 +0x16c\r\n==================\r\n--- FAIL: TestGetInstanceIDByNodeName (0.00s)\r\n\ttesting.go:699: race detected during execution of test\r\nFAIL\r\n\r\n```\r\n\r\n",
  "closed_at": "2018-01-16T14:42:33Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/13653959?v=4",
    "events_url": "https://api.github.com/users/k8s-github-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-github-robot/followers",
    "following_url": "https://api.github.com/users/k8s-github-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-github-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-github-robot",
    "id": 13653959,
    "login": "k8s-github-robot",
    "node_id": "MDQ6VXNlcjEzNjUzOTU5",
    "organizations_url": "https://api.github.com/users/k8s-github-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-github-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-github-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-github-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-github-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-github-robot"
  },
  "comments": 5,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/58325/comments",
  "created_at": "2018-01-16T08:49:59Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/58325/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/58325",
  "id": 288826684,
  "labels": [
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a bug.",
      "id": 105146071,
      "name": "kind/bug",
      "node_id": "MDU6TGFiZWwxMDUxNDYwNzE=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/bug"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/58325/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWUyODg4MjY2ODQ=",
  "number": 58325,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "azure TestGetInstanceIDByNodeName data race",
  "updated_at": "2018-01-16T14:42:33Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/58325",
  "user": {
    "avatar_url": "https://avatars3.githubusercontent.com/u/13374016?v=4",
    "events_url": "https://api.github.com/users/hzxuzhonghu/events{/privacy}",
    "followers_url": "https://api.github.com/users/hzxuzhonghu/followers",
    "following_url": "https://api.github.com/users/hzxuzhonghu/following{/other_user}",
    "gists_url": "https://api.github.com/users/hzxuzhonghu/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/hzxuzhonghu",
    "id": 13374016,
    "login": "hzxuzhonghu",
    "node_id": "MDQ6VXNlcjEzMzc0MDE2",
    "organizations_url": "https://api.github.com/users/hzxuzhonghu/orgs",
    "received_events_url": "https://api.github.com/users/hzxuzhonghu/received_events",
    "repos_url": "https://api.github.com/users/hzxuzhonghu/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/hzxuzhonghu/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/hzxuzhonghu/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/hzxuzhonghu"
  }
}